### ISSUE TITLE
Kernel selector does not indicate that I'm currently connected to my Jupyter session

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode-jupyter#8797]
Testing #8778

After performing the steps in the test plan, if I open the kernel selector, I see the following:

![image](https://user-images.githubusercontent.com/30305945/151015333-4b88dcb7-5b43-4530-8c06-99f898d8315a.png)

It says I am currently connected to the remote kernel that I selected, but there is also a Jupyter session entry with 1 existing connection, and that does not say that I'm currently connected to that session. If anything, I'd expect the kernel selector list to say that that is my active connection.

**Comment by DonJayamanne** [comment]
I agree, however this is how VS Code API works, when you reload VS Code the connection will be restored correctly and you'll see the right kernel marked as the currently selected kernel.

We've looked into this and we don't think this can be resolved. 
VS Code treats controllers as the actual kernel & stores the mappings, when in reality in jupyter these things are like kernel specifications which could spanw a real kernel that can get listed in the dropdown.

TLDR - I don't think we can do anything about this, we've look into this.
Upon reloading VS Code the correct kernel is displayed as selected.

**Comment by greazer** [comment]
The known use cases for how this bug could cause an issue doesn't seem worth fixing for now.
